flows/engine/session.go

```diff
@@ -225,7 +232,7 @@ func (s *session) continueUntilWait(currentRun flows.FlowRun, destination flows.
 
 		// if we now have a destination, go there
 		if destination != noDestination {
-			if !visited[destination] {
+			if !s.flowStack.hasVisited(destination) {
```

Comment:
Shouldn't we have some kind of error or log when we are stopping execution due to a loop?